packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -63,21 +70,56 @@ test.describe('Chat Composite E2E Tests', () => {
     await browser.close();
   });
 
-  test('composite loads correctly with participant list', async () => {
-    await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].waitForSelector(dataUiId(IDS.participantList));
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('1-chat-screen.png');
+  test('composite pages load completely', async () => {
+    for (const idx in pages) {
+      await compositeLoaded(pages[idx]);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`1-page-${idx}-chat-screen.png`);
+    }
   });
 
   test('can send message', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
+    await page.keyboard.press('Enter');
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('2-send-message.png');
+  });
+
+  test('can receive message', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
```

Comment:
I'm confused by this test.

* We're waiting for `delivered` status to test if the message is received.
* Who sent the message? I expected us to send a message from one page and then check that it's received on the other.

Reply:
If you notice, the message was sent in the previous test by page[0]
and is being checked here by page[1]

Reply:
page[0] sent the message in a previous test.
page[1] is checking to see if it shows on their screen

Reply:
Oh... these tests aren't independent?

Is there a way to make them isolated? Or is it prohibitive in terms of runtime? Tests that aren't isolated can be flaky / one test can lead to failures for all the tests downstream making debugging much harder.

Reply:
I read playwright docs and now I understand this better.

My recommendation is: use `test.beforeAll` to start the browser and the app server. Then, use `test.beforeEach` to create a new browser context and new users and threads for each test. This way they stay isolated behaviorally.

Reply:
Running each test in isolation means spawning a new browser page and initializing it for every test. Increasing runtime.
Although, I myself want to make the tests isolated. Not 100% sure if it's worth it right now. Thoughts?

Reply:
Plus deeper scenarios will require a lot of logic duplication that other tests already tested. Any suggestions for avoiding that?

Reply:
As discussed IRL, let's land this PR with sequential tests, making it more obvious that the tests need to run sequentially. We should take another look at this before we can add the tests to CI.

Reply:
jest by default tries to run tests in parallel, is there something in the config (`--runinband`?) to ensure these run sequentially? (I can't remember if its tests suites are run in parallel or includes tests within a suite)

Reply:
These tests are not being run by jest. Playwright has its own test runner  and we are using that. It runs test in sequence.